plugin/pkg/scheduler/schedulercache/node_info.go

```diff
@@ -153,6 +165,18 @@ func calculateResource(pod *api.Pod) (cpu int64, mem int64, non0_cpu int64, non0
 	return
 }
 
+// Sets the overall node information.
+func (n *NodeInfo) SetNode(node *api.Node) error {
+	n.node = node
+	return nil
+}
+
+// Removes the overall information about the node.
+func (n *NodeInfo) RemoveNode(node *api.Node) error {
```

Comment:
I think you don't understand. If the node is deleted, there shouldn't by any pods assigned to in in db.
But we don't controller watches - they may be delivered in different orders, because those are different watches.
It's not a bug- it's like distributed system work.